### PR TITLE
Allow for getting world name and path separately on the command line

### PR DESCRIPTION
**Pull request by BitOfAByte**
Adds `--worldpath` option, which works almost the same as `--world`, except when passed `--worldpath list` it gives back paths only without quotes.

 `--worldname list` works the same as `--world list`, but only prints the name.

Inside of `main.c`, `list_worlds` and `print_worldspecs` now take booleans `print_name` and `print_path` (optional in `print_worldspec`, defaults to both) to allow for greater extensible in printing in general. If only path or only name is being printed, it does not use quotes. If both are being printed, it uses the same rules as before.

This implements #6452

### CODE REVIEW

**Comment by SmallJoker** [comment]
Sounds good, a complement for `--worldname`. Would it perhaps be possible for you to bring `--world` into an usable format as well? We should try to get the single quotation marks away and use an unique separator (tabulator was suggested).

**Comment by BitOfAByte** [comment]
That's pretty easy, it just comes down to how you want it to look. Here is how a set of worlds I have looks with the current setup:
```
$ ./bin/minetest --world list
Available worlds:
  worldname2     '/home/bitofabyte/code/minetest/bin/../worlds/worldname2'
  backslash\nn   '/home/bitofabyte/code/minetest/bin/../worlds/backslash\nn'
  'My "cool" world2' '/home/bitofabyte/code/minetest/bin/../worlds/My "cool" world2'
  worldname1     '/home/bitofabyte/code/minetest/bin/../worlds/worldname1'
  'world space'  '/home/bitofabyte/code/minetest/bin/../worlds/world space'
  'really long ' name' '/home/bitofabyte/code/minetest/bin/../worlds/really long ' name'
  'My "cool" world' '/home/bitofabyte/code/minetest/bin/../worlds/My "cool" world'
  3x3            '/home/bitofabyte/code/minetest/bin/../worlds/3x3'
```

Here is how it looks with 1 tab:

```
$ ./bin/minetest --world list
Available worlds:
        worldname2      /home/bitofabyte/code/minetest/bin/../worlds/worldname2
        backslash\nn    /home/bitofabyte/code/minetest/bin/../worlds/backslash\nn
        My "cool" world2        /home/bitofabyte/code/minetest/bin/../worlds/My "cool" world2
        worldname1      /home/bitofabyte/code/minetest/bin/../worlds/worldname1
        world space     /home/bitofabyte/code/minetest/bin/../worlds/world space
        really long ' name      /home/bitofabyte/code/minetest/bin/../worlds/really long ' name
        My "cool" world /home/bitofabyte/code/minetest/bin/../worlds/My "cool" world
        3x3     /home/bitofabyte/code/minetest/bin/../worlds/3x3
```

And here is how it looks with 2 tabs:
```
$ ./bin/minetest --world list
Available worlds:
        worldname2              /home/bitofabyte/code/minetest/bin/../worlds/worldname2
        backslash\nn            /home/bitofabyte/code/minetest/bin/../worlds/backslash\nn
        My "cool" world2                /home/bitofabyte/code/minetest/bin/../worlds/My "cool" world2
        worldname1              /home/bitofabyte/code/minetest/bin/../worlds/worldname1
        world space             /home/bitofabyte/code/minetest/bin/../worlds/world space
        really long ' name              /home/bitofabyte/code/minetest/bin/../worlds/really long ' name
        My "cool" world         /home/bitofabyte/code/minetest/bin/../worlds/My "cool" world
        3x3             /home/bitofabyte/code/minetest/bin/../worlds/3x3
```

If there's a particular style that is agreed on, I can add it to this pull request or make another.

**Comment by SmallJoker** [comment]
@BitOfAByte The last style with two tabs separates the name and the path nicely, even with long and short names. \</opinion>

File: src/main.cpp [hunk 1]
```diff
@@ -253,8 +265,10 @@ static void set_allowed_options(OptionList *allowed_options)
 			_("Same as --world (deprecated)"))));
 	allowed_options->insert(std::make_pair("world", ValueSpec(VALUETYPE_STRING,
 			_("Set world path (implies local game) ('list' lists all)"))));
+	allowed_options->insert(std::make_pair("worldpath", ValueSpec(VALUETYPE_STRING,
+			_("Set world path (implies local game) ('list' lists all)"))));
 	allowed_options->insert(std::make_pair("worldname", ValueSpec(VALUETYPE_STRING,
-			_("Set world by name (implies local game)"))));
+			_("Set world by name (implies local game) ('list' lists all)"))));
```
Comment (rubenwardy) [inline src/main.cpp]: > lists all names, no paths

File: src/main.cpp [hunk 2]
```diff
@@ -336,24 +350,33 @@ static void list_game_ids()
...
-		name = padStringRight(name, 14);
-		os << "  " << name << " " << path << std::endl;
+		if (print_name && print_path) {
+			os << "\t" << name << "\t\t" << path << std::endl;
+
+			if (name.find(' ') != std::string::npos)
+				name = std::string("'").append(name).append("'");
+			path = std::string("'").append(path).append("'");
+
+			name = padStringRight(name, 14);
```
Comment (SmallJoker) [inline src/main.cpp]: Why do you modify `name` and `path` here without using them?

Comment (BitOfAByte) [reply src/main.cpp]: Oops. I hadn't meant for all of those changes to be included.

That was when I was trying different things with spacing, so I was modifying the format a lot. I was keeping the code there so it was easy to switch formats for testing. 

If we want to use the tab formatting (which we were talking about above), that code can be removed. If we don't want to use it, I can undo the changes.

**Comment by Megaf** [comment]
Can't we just have --world-name that will always assume a world inside the default world pat and --world-path that will point to a specific world path? And list option to --world-name list available worlds.
And removing all other world things.
Also, making world-name create a world with that name if it doesn't exist. Maybe we could have the flag --create-world.

**Comment by BitOfAByte** [comment]
@Megaf that is essentially what this pull request does (except for the part about creating a world if it doesn't exist, but I want to get to that at some point in the future).

```
$ ./bin/minetest --world list
Available worlds:
        name with space         /home/bitofabyte/code/minetest_brian/bin/../worlds/name with space
        asdfjkl         /home/bitofabyte/code/minetest_brian/bin/../worlds/asdfjkl
        3x3             /home/bitofabyte/code/minetest_brian/bin/../worlds/3x3

$ ./bin/minetest --worldpath list
Available worlds:
  /home/bitofabyte/code/minetest_brian/bin/../worlds/name with space
  /home/bitofabyte/code/minetest_brian/bin/../worlds/asdfjkl
  /home/bitofabyte/code/minetest_brian/bin/../worlds/3x3

$ ./bin/minetest --worldname list
Available worlds:
  name with space
  asdfjkl
  3x3
```

The alternative is what @rubenwardy is suggesting, where there is one list command (`--worldlist name`/`--worldlist path`/`--worldlist both`) and then a `--world` and `--worldname` which only specify world names, which would make it possible to start a world that is named "list". I'm not 100% sure, but I think I might like the second (rubenwardy's) design better.

**Comment by SmallJoker** [comment]
rubenwardy's suggestion is quite good, as conflicts won't happen anymore that way.
Please don't let this PR rot away. We have enough issues without a PR as solution.

**Comment by BitOfAByte** [comment]
If that's what you want to do I'll update it soon. I think I actually wrote that commit but never pushed it. I'll look into it in the next few days, I just wasn't sure exactly what was supposed to happen here.

File: src/main.cpp [hunk 3]
```diff
@@ -336,24 +344,26 @@ static void list_game_ids()
...
-			name = std::string("'").append(name).append("'");
-		path = std::string("'").append(path).append("'");
-		name = padStringRight(name, 14);
-		os << "  " << name << " " << path << std::endl;
+		if (print_name && print_path) {
+			os << "\t" << name << "\t\t" << path << std::endl;
+		} else if (print_name) {
+			os << "  " << name << std::endl;
+		} else if (print_path) {
+			os << "  " << path << std::endl;
```
Comment (SmallJoker) [inline src/main.cpp]: `"\t" << path` and above `" \t" << name` should be used for consistency.
Either that or `" " << name` for the first case. Both is fine by me.

File: src/main.cpp [hunk 4]
```diff
@@ -252,9 +258,11 @@ static void set_allowed_options(OptionList *allowed_options)
 	allowed_options->insert(std::make_pair("map-dir", ValueSpec(VALUETYPE_STRING,
 			_("Same as --world (deprecated)"))));
 	allowed_options->insert(std::make_pair("world", ValueSpec(VALUETYPE_STRING,
-			_("Set world path (implies local game) ('list' lists all)"))));
+			_("Set world path (implies local game)"))));
 	allowed_options->insert(std::make_pair("worldname", ValueSpec(VALUETYPE_STRING,
 			_("Set world by name (implies local game)"))));
+	allowed_options->insert(std::make_pair("worldlist", ValueSpec(VALUETYPE_STRING,
+			_("Get list of worlds (implies local game) ('path' lists paths, 'name' lists names, 'both' lists both)"))));
```
Comment (SmallJoker) [inline src/main.cpp]: Line is longer than 80 characters. Please split it.

**Comment by SmallJoker** [comment]
:+1:
@rubenwardy Could you please refresh your review on this PR?

**Pull request merged**